### DISCUSSION TITLE
Developing notebooks in vscode running in the browser

### DISCUSSION THREAD

**Discussion by greazer** [microsoft/vscode-jupyter#10315] [Announcements]
## Can't install VS Code? No problem!

Editing Jupyter notebooks in VS Code can be done entirely through a web based interface provided by https://vscode.dev or https://github.dev (available by typing '.' when viewing a repo on https://github.com).

## Open a notebook
Once you choose one of the above paths, open a Jupyter notebook (.ipynb). From there, be sure to install the Jupyter and Python extensions (Python only required if using... ahem... Python).

## Run a cell 
In order to run cells in your notebook, you'll need to specify a kernel to run against. There are at least 3 ways to accomplish this:

* [Connect to a Jupyter server](https://github.com/microsoft/vscode-jupyter/wiki/Connecting-to-a-remote-Jupyter-server-from-vscode.dev)
* [Install Pyolite](https://marketplace.visualstudio.com/items?itemName=joyceerhl.vscode-pyodide) (Jupyter extension not required!)
* [Github Codespace](https://github.com/features/codespaces) If you opened a Github repo, you may have the option to create and connect to a Github Codespace. Run a cell and you'll be prompted appropriately.

### Connecting to a Jupyter server or running with the Pyolite kernel
Most features will operate just fine; however, we are still working to support the following:
- Debugging in an Interactive Window session
- Running local kernels/python environments (you have to start your own jupyter instead)
- Intellisense is limited
- Dataframe viewing
- Plot expansion

### Connecting to a Github Codespace
All functionality is available

## Tell us what you think!
Feel free to add comments below, [enter a New Issue](https://github.com/microsoft/vscode-jupyter/issues/new?assignees=&labels=bug&template=1_bug_report.md) or [start a new GitHub Discussion](https://github.com/microsoft/vscode-jupyter/discussions)

**Comment by DmitriyG228** [comment]
Sorry, I am trying to understand what is going on. That is in order  to connect to a running Jupyter server, rather than running kernels with vscode, right? If so, that would allow keeping kernels running after disconnecting from vscode, which is interesting. If so, I still do not get it how to make it working, as "Editing Jupyter notebooks in VS Code can be done entirely through a web based interface provided by https://vscode.dev/ or https://github.dev/ (available by typing '.' when viewing a repo on https://github.com/)." does not make sense to me. Thank you!
